app/controllers/avo/search_controller.rb

```diff
@@ -87,18 +87,24 @@ def apply_attach_scope(query, parent)
       # In these scenarios, try to find the grandparent for the new views where the parent is nil
       # and initialize the parent record with the grandparent attached so the user has the required information
       # to scope the query.
+      # Example usage: Got to a project, create a new review, and search for a user.
       if parent.blank? && params[:via_parent_resource_id].present? && params[:via_parent_resource_class].present? && params[:via_relation].present?
-        grandparent = params[:via_parent_resource_class].safe_constantize.find params[:via_parent_resource_id]
-        parent = params[:via_reflection_class].safe_constantize.new(
+        parent_resource_class = BaseResource.valid_model_class params[:via_parent_resource_class]
+        reflection_class = BaseResource.valid_model_class params[:via_reflection_class]
+
+        grandparent = parent_resource_class.find params[:via_parent_resource_id]
+        parent = reflection_class.new(
           params[:via_relation] => grandparent
```

Comment:
This `params[:via_relation]` is not ok. It should be something like `reviewable`, not a model class.